### ISSUE TITLE
Server hangs if client connection not explicitly closed

### ISSUE THREAD

**Issue by lutusp** [AlexandreRouma/SDRPlusPlus#1358]
This may be taken as a feature request rather than a bug report (it's both), but if a client disconnects from an SDR++ server without first closing the network connection, the server doesn't know it's been orphaned and refuses new connections, then any new clients crash.

One remedy would be to automatically close any active network connection when the user closes the app (this would free the server). Another would be to prevent the client crash, instead print the connection failure reason on stdout and carry on without crashing. The first remedy is better overall.

Another remedy would be for the client/server connection to use a heartbeat signal to detect active connections, but this is more difficult to implement.

Also ... it's a terrific program. I know, I've written a few myself and this one is much better.

**Comment by AlexandreRouma** [comment]
Yeah I'll switch this a to a feature request as the server protocol doesn't have a heartbeat system. This will be taken care of in version 2.0 of the protocol.
